test/e2e/framework/util.go

```diff
@@ -733,6 +733,39 @@ func WaitForPodsRunningReady(c clientset.Interface, ns string, minPods, allowedN
 	return nil
 }
 
+// WaitForDaemonSets for all daemonsets in the given namespace to be ready
+// (defined as all but 'allowedNotReadyNodes' pods associated with that
+// daemonset are ready).
+func WaitForDaemonSets(c clientset.Interface, ns string, allowedNotReadyNodes int32, timeout time.Duration) error {
```

Comment:
This expects that all daemonsets have been applied and that the ds controller has synced. Are we concerned about a race?

Reply:
This code is running fairly late (when all nodes become ready), so I think that there is no real risk of raciness.

If we see that it happens in test runs, I will add some additional logic here.